src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExecuteActionTests.cs

```diff
@@ -0,0 +1,132 @@
+
+            InvalidOperationException invalidOperationException = await Assert.ThrowsAsync<InvalidOperationException>(
+                () => action.ExecuteAsync(options, null, CreateCancellationToken()));
+
+            Assert.Contains(string.Format(Strings.ErrorMessage_NonzeroExitCode, "-1"), invalidOperationException.Message);
+        }
+
+        [Fact]
+        public async Task ExecuteAction_TokenCancellation()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "TokenCancellation" }); ;
+
+            TaskCanceledException taskCanceledException = await Assert.ThrowsAsync<TaskCanceledException>(
+                () => action.ExecuteAsync(options, null, CreateCancellationToken()));
+
+            Assert.Contains(TaskCanceledMessage, taskCanceledException.Message);
+        }
+
+        [Fact]
+        public async Task ExecuteAction_TextFileOutput()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            DirectoryInfo outputDirectory = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "ExecuteAction", Guid.NewGuid().ToString()));
+            string textFileOutputPath = outputDirectory.FullName + "\\file.txt";
+
+            const string testMessage = "TestMessage";
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "TextFileOutput", textFileOutputPath, testMessage });
+
+            CollectionRuleActionResult result = await action.ExecuteAsync(options, null, CreateCancellationToken());
+
+            Assert.Equal("0", result.OutputValues["ExitCode"]);
+            Assert.Equal(testMessage, File.ReadAllText(textFileOutputPath));
+
+            try
+            {
+                outputDirectory?.Delete(recursive: true);
+            }
+            catch
+            {
+            }
+        }
+
+        [Fact]
+        public async Task ExecuteAction_InvalidPath()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            string uniquePathName = Guid.NewGuid().ToString();
+
+            options.Path = uniquePathName;
+            options.Arguments = GenerateArgumentsString(Array.Empty<string>());
+
+            FileNotFoundException fileNotFoundException = await Assert.ThrowsAsync<FileNotFoundException>(
+                () => action.ExecuteAsync(options, null, CreateCancellationToken()));
+
+            Assert.Equal(string.Format(Strings.ErrorMessage_FileNotFound, uniquePathName), fileNotFoundException.Message);
+        }
+
+        private static string GenerateArgumentsString(string[] additionalArgs)
+        {
+            return Assembly.GetExecutingAssembly().Location.Replace(
```

Comment:
Please use AssemblyHelper.GetAssemblyArtifactBinPath to get the path to the assembly.